code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,2046 @@
+		alpha = 255
+		speed = 1
+
+/mob/living/simple_animal/hostile/serf/proc/handlecloak()
+	var/turf/simulated/T = get_turf(src)
+	var/light_available = T.get_lumcount(0.5) * 10
+
+	if(cloaked)
+		if(!istype(T))
+			return
+		if(light_available <= 1)
+			alpha = round((255 * 0.35))
+			return
+		else
+			alpha = round((255 * 0.80))
+	else
+		return
+
+/obj/effect/dummy/serf
+	name = "???"
+	desc = "Something is moving underground..."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	density = 0
+	anchored = 1
+	invisibility = 60
+	burn_state = LAVA_PROOF
+
+/obj/effect/dummy/serf/relaymove(mob/user, direction)
+	forceMove(get_step(src,direction))
+	user.dir = direction
+
+/obj/effect/dummy/serf/ex_act()
+	return
+
+/obj/effect/dummy/serf/bullet_act()
+	return
+
+/obj/effect/dummy/serf/singularity_act()
+	return
+
+/mob/living/simple_animal/hostile/serf/death()
+	. = ..()
+	if(!.)
+		return FALSE
+
+	var/turf/E = get_turf(src)
+	var/atom/movable/overlay/animation = new /atom/movable/overlay(E)
+	qdel(src)
+	animation.name = "???"
+	animation.density = 0
+	animation.anchored = 1
+	animation.icon = 'icons/mob/mob.dmi'
+	animation.icon_state = "[serftype]_serf_die"
+	animation.layer = MOB_LAYER
+	animation.desc = "What..."
+	animation.master = holder
+	animation.dir = dir
+	flick("[serftype]_serf_die",animation)
+	sleep(14)
+	if(animation)
+		qdel(animation)
+
+/mob/living/simple_animal/hostile/serf/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Time survived: [life].")
+
+/mob/living/simple_animal/hostile/serf/Life()
+	. = ..()
+	if(stat != DEAD)
+		life++
+		dissolve_cooldown--
+		appear_cooldown--
+		glare_cooldown--
+		update_type()
+		if(serftype == "cling")
+			ventcrawler = TRUE
+			speed = 0
+		if(serftype == "sling")
+			handlecloak()
+			if(!slingabilities)
+				var/datum/action/innate/serf/glare/glare_action = new()
+				glare_action.Grant(src)
+				var/datum/action/innate/serf/cloak/cloak_action = new()
+				cloak_action.Grant(src)
+				slingabilities = 1
+		if(life < 600) // Newly spawned
+			maxHealth = 150
+			desc = "An impossible being that breaks all concepts of reality just by existing."
+			melee_damage_lower = 20
+			melee_damage_upper = 25
+			color = ""
+			environment_smash = 1
+		if(life >= 600) // Been alive for some time, let's give you more health.
+			maxHealth = 200
+			desc = "An impossible being that breaks all concepts of reality just by existing. This one has existed for some time."
+			melee_damage_lower = 20
+			melee_damage_upper = 25
+			color = ""
+			environment_smash = 1
+		if(life >= 1200) // How the heck are you still alive. Let's make it easier for you to survive, then.
+			maxHealth = 250
+			desc = "An impossible being that breaks all concepts of reality just by existing. This one has existed for a long time."
+			melee_damage_lower = 25
+			melee_damage_upper = 35
+			color = ""
+			environment_smash = 1
+		if(life >= 2000) // Last upgrade.
```

Comment:
Checking what's their life number multiple time everytime is again, horrifyingly inefficient.

This should also be an else if chain or a switch statement. If life < 600, then it couldn't meet any of the other conditions below and therefore checking them is unnecessary.